kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -91,4 +96,25 @@ static Map<ResourceID, KafkaClusterRef> clusterRefs(Context<KafkaProxy> context)
                 .collect(Collectors.toMap(ResourceID::fromResource, Function.identity()));
     }
 
+    public static String name(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getName();
+    }
+
+    public static String namespace(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getNamespace();
+    }
+
+    public static Long generation(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getGeneration();
+    }
+
+    public static String uid(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getUid();
+    }
+
+    public static <T extends HasMetadata> Optional<T> resourceNamed(@NonNull Collection<T> collection, @NonNull String name) {
```

Comment:
nit: personally I'd swap the arg order of name and collection and relatedly is it worth having a var args version `resourceNamed(String name, T... existing)`?

Reply:
A varargs version might become useful with a refactor to the ITs, but wouldn't add it speculatively.